modules/standard/MemMove.chpl

```diff
@@ -51,10 +51,32 @@ module MemMove {
     :return: ``true`` if ``t`` needs to be deinitialized
     :rtype: param bool
   */
+  pragma "no doc"
+  deprecated "'needsDeinit' is deprecated; please use 'needsDestroy' instead"
   proc needsDeinit(type t) param {
     return __primitive("needs auto destroy", t);
   }
 
+  /*
+    Check to see if a given type would normally be destroyed automatically
+    by the compiler when its lifetime ends. For example, passing an ``owned``
+    class would see this function return ``true``, whereas passing an
+    ``unmanaged`` class would result in ``false``.
+
+    A ``shared`` class would also result in ``true`` because the individual
+    managed instance would be destroyed, but not necessarily the actual object
+    being managed.
```

Comment:
Can we reword it? Here is a suggestion:

> A ``shared`` class would also result in ``true`` because the compiler destroys ``shared`` variables to decrements the reference count (and possibly the object being managed, as well).